### PR TITLE
W-15176571: Compatibility with J17 (And corresponding annotation)

### PR DESCRIPTION
**Pull request by anitacruz**
Tested against MUnit 3.3.0-SNAPSHOT at https://jenkins-onprem.build.msap.io/job/Mule-runtime/job/extensions/job/mule-aggregators-module/job/PR-199/11/
J17 compatibility changes

_Hi, thank you for your work. We understand that you want to merge your changes and move on from this issue, but we also want to maintain the safety, readability, and testability of our code. Because of this, we kindly ask that you confirm that you have fulfilled the prerequisites for each category of activity before merging your changes._

### If the Pull Request has a dependency update:

- [ ] I have read the Release Notes for the new version (and intermediate versions if the jump would include more than one). _Don't blindly trust that the dependency will honor SEMVER, always read carefully the release notes_
- [ ] Java 8+ support is maintained
- [ ] The new version has no vulnerabilities
- [ ] A team member reviewed the Pull Request.
- [ ] No backward compatibility is broken.
- [ ] If it is a "provided" dependency, it has been checked that if it is suggested, the version number is the same.

### For ALL the Releases:

- [ ] There is a task in GUS for this change.
- [ ] The corresponding Release Notes have been written and shared with the documentation team.
- [ ] The new code has tests and they have before and after methods to be sure that you are working with a clean environment and that you are leaving the environment clean after they finish.
- [ ] If you are using reflection, create a test to call the methods you are invoking, this way, if there is a change in the API, the test will be able to detect it.
- [ ] If the project has a parent pom and modules, the release pipeline only uploads the modules, so, please, make sure that the parent was deployed in the repositories before considering this release finished.
- [ ] Notify in the Slack Channel that the release is complete, so the Docs team can merge the Documentation and Release Notes PR. The docs team always checks in Anypoint Exchange that the connector is released before merging the release notes.
- [ ] Add the connector build to GUS, and assign that build to the GUS ticket.
- [ ] Create a change case request, based on the Change Case Management doc https://docs.google.com/document/d/1tMJlTTZLaXMLiYwxlMBFY1yJwBmejhc4-IP8HAXgDfY/edit#

**NOTE**: _(applies only for Core-Connectors connectors and modules) The release process ends when you merge the pull request that updates the support branch to the new snapshot, please don't forget to do this._

### Patch Version:

- [ ] The Pull Request has been peer-reviewed.
- [ ] No backward compatibility is broken.
- [ ] Documentation: (Required) Release Notes PR with compatibility table. Share it with the Docs team.

### CODE REVIEW

File: build.yaml [hunk 1]
```diff
@@ -1 +1,5 @@
 projectType: extension
+additionalConfigs:
+  jdk17:
+    testJdkTool: OPEN-JDK17
+    mavenAdditionalArgs: -DminVersion=4.6.0 -Dtest=none -DfailIfNoTests=false -DruntimeProduct=MULE_EE
```
Comment (acurci-at-mulesoft) [inline build.yaml:5]: Couldn't you add additional configs for JDK8 and JDK11 (both of them with -DminVersion=4.3.0)?

Comment (anitacruz) [reply build.yaml:5]: Definitely! Didn't know if I should or not. Will do it right away

File: pom.xml [hunk 2]
```diff
@@ -79,8 +82,9 @@
                     </argLines>
                     <runtimeConfiguration>
                         <discoverRuntimes>
-                            <product>ALL</product>
-                            <includeSnapshots>true</includeSnapshots>
+                            <product>EE</product>
```
Comment (acurci-at-mulesoft) [inline pom.xml:85]: Why are we setting the enterprise editions?

**Review by acurci-at-mulesoft — DISMISSED** [review]

File: pom.xml (hunk 2 above)
Comment (acurci-at-mulesoft) [reply pom.xml:84]: It's ok. I was just wondering why we were testing with EE when a lot of connectors should work with the CE version.

**Comment by acurci-at-mulesoft** [comment]
@anitacruz please, check the Renovate PRs that are related to the dependency updates you made, and close them.

**Comment by anitacruz** [comment]
The changes necessary for the connector to be compatible with Java 17 (including adding the annotations) are ready and pushed in this branch: https://github.com/mulesoft/mule-aggregators-module/tree/chore/W-15176571. But, thanks to our work in this and a test added while increasing the coverage, we managed to find a bug in the runtime regarding an illegal reflective access with dataweave. 

The fix is included in version 4.6.3-SNAPSHOT, which will be released as a stable version on the next monthly release, which is on June 4th, as discussed in the following thread: https://salesforce-internal.slack.com/archives/C79H0F9HC/p1713989449125389. 

Therefore, the connector can't be released as compatible with Java 17 until this fix is released on the runtime's side.

**Review by juanfloresMule — APPROVED** [review]

**Pull request merged**